ml_service/util/manage_environment.py

```diff
@@ -22,14 +24,47 @@ def get_environment(
                 restored_environment = environments[environment_name]
 
         if restored_environment is None or create_new:
-            new_env = Environment.from_conda_specification(
-                environment_name,
-                os.path.join(e.sources_directory_train, conda_dependencies_file),  # NOQA: E501
-            )  # NOQA: E501
-            restored_environment = new_env
+
+            if restored_environment is None:
+                # Environment has to be created
+                if conda_dependencies_file is not None:
+                    new_env = Environment.from_conda_specification(
+                        environment_name,
+                        os.path.join(e.sources_directory_train, conda_dependencies_file),  # NOQA: E501
+                    )  # NOQA: E501
+                    restored_environment = new_env
+                else:
+                    restored_environment = Environment(environment_name)
+            else:
+                if conda_dependencies_file is not None:
+                    # Environment has to be updated
+                    restored_environment.conda_dependencies_file =\
+                        os.path.join(e.sources_directory_train,
+                                     conda_dependencies_file)
+
             if enable_docker is not None:
                 restored_environment.docker.enabled = enable_docker
-                restored_environment.docker.base_image = DEFAULT_GPU_IMAGE if use_gpu else DEFAULT_CPU_IMAGE  # NOQA: E501
+
+                if docker_image is not None:
+                    restored_environment.docker.base_image = docker_image
+                    # In case of own image
+                    # don't append AML managed dependencies
+                    restored_environment.python.\
+                        user_managed_dependencies = True
+                elif dockerfile is not None:
+                    # Alternatively, load from a file.
+                    with open(dockerfile, "r") as f:
+                        dockerfile = f.read()
+                        restored_environment.docker.\
+                            base_dockerfile = dockerfile
+                    # In case of own Dockerfile
+                    # don't append AML managed dependencies
+                    restored_environment.python.\
+                        user_managed_dependencies = True
```

Comment:
Setting this option is relevant to not mix up dependencies and conda environments within the new docker image created. When I didn't set this the mounting of the input data with `fuse` inside the container failed.